casper/src/main/scala/coop/rchain/casper/api/BlockAPI.scala

```diff
@@ -70,6 +71,13 @@ object BlockAPI {
         _ <- triggerPropose.traverse(_(casper, true))
       } yield r
 
+    // Check if node is read-only
+    val readOnlyError = new RuntimeException(
+      s"Deploy was rejected because node is running in read-only mode. " +
+        s"To be a validator node should be run with a validator private key."
```

Comment:
This error message is also returned to the client via API so only the first sentence makes sense.

With API rewrite we need to distinguish debug messages and friendly messages returned to the end user.

Reply:
Yes, I agree. Fixed.